### PR TITLE
Check NR state validity

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
We don't check state validity after NR, and we sometimes update network buses voltage with negative values

**What is the new behavior (if this is a feature change)?**
A new status UNREALISTIC_STATE has been added

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/ac/nr/NewtonRaphsonStatus.java [hunk 1]
```diff
@@ -13,5 +13,6 @@ public enum NewtonRaphsonStatus {
     CONVERGED,
     MAX_ITERATION_REACHED,
     SOLVER_FAILED,
-    NO_CALCULATION
+    NO_CALCULATION,
+    INVALID_SOLUTION
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/ac/nr/NewtonRaphsonStatus.java]: INVALID is not the right term for me, it is more NON_PHYSICAL

File: src/main/java/com/powsybl/openloadflow/ac/nr/NewtonRaphson.java [hunk 2]
```diff
@@ -182,6 +179,22 @@ public void updateNetwork() {
...
+        for (Variable<AcVariableType> v : equationSystem.getIndex().getSortedVariablesToFind()) {
+            if (v.getType() == AcVariableType.BUS_V) {
+                double value = equationSystem.getStateVector().get(v.getRow());
+                if (value < parameters.getMinRealisticVoltage() || value > parameters.getMaxRealisticVoltage()) {
+                    busesOutOfNormalVoltageRange.add(network.getBus(v.getElementNum()).getId());
+                }
+            }
+        }
+        if (!busesOutOfNormalVoltageRange.isEmpty()) {
+            LOGGER.warn("{} buses have a voltage magnitude out of range [0.5, 1.5]: {}", busesOutOfNormalVoltageRange.size(), busesOutOfNormalVoltageRange);
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/ac/nr/NewtonRaphson.java:193]: You should change the error message.

**Review by annetill — APPROVED** [review]

**Comment by geofjamg** [comment]
@annetill is 0.5 - 1.5 ok ?

**Comment by annetill** [comment]
> @annetill is 0.5 - 1.5 ok ?

Maybe yes.

**Pull request merged**